.github/workflows/gh-pages.yml

```diff
@@ -0,0 +1,33 @@
+name: github pages
+
+on:
+  push:
+    branches:
+      - main # Set a branch name to trigger deployment
+
+jobs:
+  deploy:
+    name: Deploy doc
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@main
+
+      - name: Use OCaml
+        uses: ocaml/setup-ocaml@v2
+        with:
+          ocaml-compiler: '5.1'
+          dune-cache: true
+
+      - name: Deps
+        run: opam install odig ambient-context ambient-context-lwt ambient-context-eio
+
+      - name: Build
+        run: opam exec -- odig odoc --cache-dir=_doc/ ambient-context ambient-context-lwt ambient-context-eio
```

Comment:
Can this be dynamically derived? as more packages are added (there's so many in branches rn ;_;)

Reply:
not that I know of, really. CI is just pain all over anyway.